kvbc/src/Replica.cpp

```diff
@@ -567,10 +572,19 @@ Replica::Replica(ICommunication *comm,
         throw std::invalid_argument{msg};
       }
     }
+
+    // A callback to report into State Transfer before blocks are being deleted
+    std::function<void(BlockId)> deleteBlocksStateTransferCb = [stateTransfer = this->m_stateTransfer](BlockId until) {
+      if (stateTransfer->isRunning()) {
+        stateTransfer->reportLastAgreedPrunableBlockId(until - 1);
+      }
+    };
+
     op_kvBlockchain.emplace(storage::rocksdb::NativeClient::fromIDBClient(m_dbSet.dataDBClient),
                             linkStChain,
                             kvbc_categories,
-                            concord::kvbc::adapter::aux::AdapterAuxTypes(this->aggregator_, this->replicaResources_));
+                            concord::kvbc::adapter::aux::AdapterAuxTypes(
+                                this->aggregator_, this->replicaResources_, std::move(deleteBlocksStateTransferCb)));
```

Comment:
I don't fully understand the need for this callback if we already have a call in ```Replica::deleteBlocksUntil``` which should be a wrapper for any call to the replica adapter.
i.e. per my understanding, we'll call ```reportLastAgreedPrunableBlockId``` twice, no?